toolchain/check/import_cpp.cpp

```diff
@@ -1550,6 +1546,54 @@ auto ImportNameFromCpp(Context& context, SemIR::LocId loc_id,
   // Access checks are performed separately by the Carbon name lookup logic.
   lookup->suppressAccessDiagnostics();
 
+  if (lookup->isOverloadedResult() ||
+      (lookup->isSingleResult() && lookup->getFoundDecl()->getAsFunction())) {
+    auto overloaded_func_decl_inst = SemIR::OverloadedCppFunctionDecl{
+        SemIR::TypeId::None, SemIR::OverloadedCppFunctionId::None};
+    auto decl_inst_id = AddPlaceholderInstInNoBlock(
+        context, Parse::NodeId::None, overloaded_func_decl_inst);
+    context.imports().push_back(decl_inst_id);
+
+    clang::UnresolvedSet<8> overload_set;
+    overload_set.append(lookup->begin(), lookup->end());
+
+    auto overloaded_function_info =
+        SemIR::OverloadedCppFunction{.name_id = name_id,
+                                     .parent_scope_id = scope_id,
+                                     .candidate_functions = overload_set};
+
+    overloaded_func_decl_inst.overloaded_function_id =
+        context.overloaded_cpp_functions().Add(overloaded_function_info);
+
+    overloaded_func_decl_inst.type_id = GetOverloadedCppFunctionType(
+        context, overloaded_func_decl_inst.overloaded_function_id,
+        SemIR::SpecificId::None);
+
+    ReplaceInstBeforeConstantUse(context, decl_inst_id,
+                                 overloaded_func_decl_inst);
+
+    // TODO: fix access for overloaded functions.
+    // For now supporting only overloaded set where all functions have the same
+    // access, in order to keep the existing functionality for single
+    // functions until the same is fixed for the overloaded functions.
+    auto overloaded_set_access = lookup->begin().getAccess();
+    for (auto i = lookup->begin() + 1, e = lookup->end(); i < e; ++i) {
```

Comment:
I believe this is a more common and clearer style:

```suggestion
    for (auto it = lookup->begin() + 1, end = lookup->end(); it != end; ++it) {
```

Reply:
Done.